### DISCUSSION TITLE
Art asset size restrictions

### DISCUSSION THREAD

**Discussion by Svarr** [sll552/DiscordBee#103] [Q&A] [answered]
The _Rich Presence_ > _Art Assets_ section of the Discord Developer Portal is mentioning a restriction of the size of art assets:
![image](https://user-images.githubusercontent.com/4354904/142827892-3203c561-cf50-4ac8-9a5b-b3a8ab09ac48.png)

Does it also apply to uploads via API? If so, I think it should be mentioned in the settings window (e.g. as a tooltip of the _Upload album covers_ checkbox) for anyone wondering why some of their album art is not uploaded / shown in their Rich Presence.

**Comment by sll552** [comment, marked as answer]
This should not be a problem because of https://github.com/sll552/DiscordBee/blob/v2.0.2/DiscordTools/AssetManager.cs#L225

I don't know if there are file size restrictions, those might be a problem if they exists.

**Reply by Svarr** [reply]
Ah, nice. 😄
But why not allowing images bigger than 1024x1024 or with non-quadratic aspect ratio?

**Reply by sll552** [reply]
Ask Discord, I guess they want to save the resources needed for serverside processing.

For DiscordBee everything is allowed, the linked `if` just decides that the image needs to be resized before uploading.
